WordPress/Classes/Services/MediaService.m

```diff
@@ -57,7 +57,10 @@ - (Media *)createMediaWith:(id<ExportableAsset>)exportable
         }
         
         if (blog.allowedFileTypes != nil) {
-            allowedFileTypes = blog.allowedFileTypes;
+            // HEIC isn't supported when uploading an image, so we filter it out (http://git.io/JJAae)
```

Comment:
As discussed, it's better to fix this client-side (iOS) for now. I confirmed that the "heic" is returned by:

- WPCom
- Jetpack
- Self-hosted sites

But in the future, the API should not return `heic` if it can't accept the file type.